cmd/batch-expire.go

```diff
@@ -534,18 +535,34 @@ func (r *BatchJobExpire) Start(ctx context.Context, api ObjectLayer, job BatchJo
 		return err
 	}
 
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	ctx, cancelCancel := context.WithCancelCause(ctx)
+	defer func() {
+		if context.Cause(ctx) == nil {
+			cancelCancel(nil)
+		}
+	}()
```

Comment:
The first error is sticky, so no need to make it complicated:
```suggestion
	defer cancelCancel(nil)
```